dashboard/app/models/pd/application/application_base.rb

```diff
@@ -0,0 +1,75 @@
+# == Schema Information
+#
+# Table name: pd_applications
+#
+#  id                  :integer          not null, primary key
+#  type                :string(255)      not null
+#  application_year    :string(255)      not null
+#  application_type    :string(255)      not null
+#  regional_partner_id :integer
+#  status              :string(255)      not null
+#  locked_at           :datetime
+#  notes               :text(65535)
+#  form_data           :text(65535)      not null
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#
+# Indexes
+#
+#  index_pd_applications_on_application_type     (application_type)
+#  index_pd_applications_on_application_year     (application_year)
+#  index_pd_applications_on_regional_partner_id  (regional_partner_id)
+#  index_pd_applications_on_status               (status)
+#  index_pd_applications_on_type                 (type)
+#
+
+# Base class for the Pd application forms.
+# Make sure to use a derived class for a specific application type and year.
+# This on its own will fail validation.
+module Pd::Application
+  class ApplicationBase < ActiveRecord::Base
+    include ApplicationConstants
+    include Pd::Form
+
+    after_initialize -> {self.status = :unreviewed}
+    after_initialize :set_type_and_year
```

Comment:
No, we don't _need_ both, but here's my reasoning:

I actually added the `after_initialize` first, because I like to have it set on a new object. 
I then later added `before_validation`, so it will always be enforced before validation/save.

Alternatively, we could set it only in `after_initialization` and then override the setters (`application_year=` & `application_type=` to raise an error or log warning and no-op.

Thoughts?

Reply:
That makes sense - I didn't think about why it would be useful for objects even before they are saved to have the values set.